### PR TITLE
Added a Fisher

### LINKED ISSUE #499: [Utility Suggestion] Fisher
When placed above water gets items from the fishing loot table over time. Luck and speed increases over time, so at LV you'll probably be getting some junk. (I personally think they shouldn't require bait)

### PR DESCRIPTION
**Pull request by BxFS**
First time doing this, sorry if my code looks terrible and/or too copypasty.
Closes #499 if accepted.

### CODE REVIEW

File: src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        return builder.build(getHolder(), entityPlayer);
+    }
+
+    public void update() {
+        super.update();
+        if (!getWorld().isRemote && energyContainer.getEnergyStored() >= energyAmountPerFish && getTimer() % fishingTicks == 0L &&
+            importItems.getStackInSlot(0).getCount() > 0) {
+            WorldServer world = (WorldServer) this.getWorld();
+            BlockPos waterCheckPos = getPos().down();
+            if (world.getBlockState(waterCheckPos).getBlock().getRegistryName().toString().equals("minecraft:water")) {
```
Comment (Archengius) [inline src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java]: would be better to check amount of water blocks, single-puddle fisher is weird imo
Also don't use registry names for block equality, use
`blockState.getBlock() instanceof BlockLiquid && blockState.getMaterial() == Material.WATER` here

File: src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                LootTable table = world.getLootTableManager().getLootTableFromLocation(new ResourceLocation("minecraft:gameplay/fishing"));
```
Comment (Archengius) [inline src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java]: Don't use string constants for resource names, there's LootTables containing resource locations of all vanilla loottables

File: src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        }
+    }
+
+    @Override
+    protected IItemHandlerModifiable createImportItemHandler() {
+        return new ItemStackHandler(1) {
+            @Nonnull
+            @Override
+            public ItemStack insertItem(int slot, @Nonnull ItemStack stack, boolean simulate) {
+                if(stack.getItem().getRegistryName().toString().equals("minecraft:string")) {
```
Comment (Archengius) [inline src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java]: same here - do not use registry name for block checks, and using Ore Dictionary would be better here
`OreDictUnifier.getOreDictNames(itemStack).contains("string")`

File: src/main/java/gregtech/common/metatileentities/MetaTileEntities.java [hunk 4]
```diff
@@ -495,6 +496,8 @@ public static void init() {
 
         TESLA_COIL = new MetaTileEntityTeslaCoil(gregtechId("tesla_coil"));
         GregTechAPI.registerMetaTileEntity(1001, TESLA_COIL);
+        FISHER = new MetaTileEntityFisher(gregtechId("fisher"));
+        GregTechAPI.registerMetaTileEntity(1050, FISHER);
```
Comment (Archengius) [inline src/main/java/gregtech/common/metatileentities/MetaTileEntities.java]: Multiple tiers would be better here, just like most of machines do

File: src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+import net.minecraftforge.items.IItemHandlerModifiable;
+import net.minecraftforge.items.ItemHandlerHelper;
+import net.minecraftforge.items.ItemStackHandler;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+
+public class MetaTileEntityFisher extends MetaTileEntity {
```
Comment (Archengius) [inline src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java]: As i said earlier, better make it tiered, that way you only need to extend TieredMetaTileEntity
and not mess with energy container yourself

File: src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        };
+    }
+
+    @Override
+    protected IItemHandlerModifiable createExportItemHandler() {
+        return new ItemStackHandler(inventorySize);
+    }
+
+    public void renderMetaTileEntity(CCRenderState renderState, Matrix4 translation, IVertexOperation[] pipeline) {
+        Textures.SOLID_STEEL_CASING.render(renderState, translation, pipeline);
```
Comment (Archengius) [inline src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityFisher.java]: for tiered machines you also don't need that

**Comment by BxFS** [comment]
Well, forgot to leave a note that I've completed required changes. Please review when you have time.

**Pull request merged**